### DISCUSSION TITLE
openapi.json and schema would not update when update enum with aenum

### DISCUSSION THREAD

**Discussion by yangyaofei** [fastapi/fastapi#8737] [Questions] [answered]
### First check

* [✅ ] I added a very descriptive title to this issue.
* [✅ ] I used the GitHub search to find a similar issue and didn't find it.
* [✅ ] I searched the FastAPI documentation, with the integrated search.
* [✅ ] I already searched in Google "How to X in FastAPI" and didn't find any information.
* [✅ ] I already read and followed all the tutorial in the docs and didn't find an answer.
* [✅ ] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* [✅ ] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* [✅ ] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* [✅ ] After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Description

I use aenum to update a Enum with fastapi, it shows as follow:

```Python
from fastapi import FastAPI
from aenum import extend_enum
from enum import Enum

app = FastAPI()

class Animal(str, Enum):
    cat = "cat"
    dog = "dog"

@app.get("/call")
def call(animal: Animal):
    return {"Hello": "World"}

@app.post("/extend_animal")
def extend_animal(animal: str):
    extend_enum(Animal, animal, animal)
    return 201
```

And when I call `extend_animal("sheep")`, the Animal will extend

But , I check the openapi.json it would not change and when I call the `call("sheep")`
it return a validation exception.  I test this. it's not a pydantic problem.

I have a way to work around it, like to refresh the openapi.json and dependant.

Should I make a pull request to add a function for FastAPI class or add a function like a tool script ?

### Environment

* OS: [Linux / macOS]:
* FastAPI Version [0.65.2]:
* Python version: 3.7.11

**Comment by YuriiMotov** [comment, marked as answer]
Openapi schema is supposed to be static (it shouldn't change during the execution of your app).
FastAPI generates schema on first request to `/openapi.json` and caches it. Any subsequent requests to `/openapi.json` will re-use cached schema
